src/CommentSold/Api/ShopClient.php

```diff
@@ -0,0 +1,331 @@
+    public function deleteWebhookListener(int $webhookId)
+    {
+        return $this->restClient->delete($this->shopToken, "{$this->shopId}/webhooks/{$webhookId}");
+    }
+
+    /**
+     * Returns a paginated list of customers
+     */
+    public function getCustomers(int $page = 1, int $perPage = Rest::PER_PAGE)
+    {
+        return $this->restClient->get($this->shopToken, "{$this->shopId}/customers?page={$page}&perPage={$perPage}");
+    }
+
+    /**
+     * Add a new customer
+     */
+    public function createCustomer(array $payload)
+    {
+        return $this->restClient->post($this->shopToken, "{$this->shopId}/customers", $payload);
+    }
+
+    /**
+     * Returns a customer for the given CommentSold customer id
+     */
+    public function getCustomer(int $customerId)
+    {
+        return $this->restClient->get($this->shopToken, "{$this->shopId}/customers/{$customerId}");
+    }
+
+    /**
+     * Update an existing customer
+     */
+    public function updateCustomer(int $customerId, array $payload)
+    {
+        return $this->restClient->post($this->shopToken, "{$this->shopId}/customers/{$customerId}", $payload);
+    }
+
+    /**
+     * Returns a paginated list of customers filtered by the search term
+     */
+    public function searchCustomers(string $searchTerm, int $page = 1, int $perPage = Rest::PER_PAGE)
+    {
+        return $this->restClient->post($this->shopToken, "{$this->shopId}/customers/search?page={$page}&perPage={$perPage}", ['term' => $searchTerm]);
+    }
+
+    /**
+     * Returns a customer for the given external customer id
+     */
+    public function getCustomerByExternalId(string $customerId)
+    {
+        return $this->restClient->get($this->shopToken, "{$this->shopId}/customers/externalId/{$customerId}");
+    }
+
+    /**
+     * Updates the address for the shop
+     */
+    public function updateShop(array $payload)
+    {
+        return $this->restClient->patch($this->shopToken, "{$this->shopId}/shop", $payload);
+    }
+
+    /**
+     * Update the products to have the provided category. The existing category for the products will be overwritten.
+     */
+    public function updateProductCategories(array $payload)
+    {
+        return $this->restClient->post($this->shopToken, "{$this->shopId}/product_categories", $payload);
+    }
+
+    /**
+     * Start allocations for dropship products
+     */
+    public function startDropshipAllocation(array $payload)
+    {
+        return $this->restClient->post($this->shopToken, "{$this->shopId}/dropshipping/allocations", $payload);
+    }
+
+    /**
+     * Restock allocations for dropship products
+     */
+    public function restockDropshipAllocation(array $payload)
+    {
+        return $this->restClient->put($this->shopToken, "{$this->shopId}/dropshipping/allocations", $payload);
+    }
+
+    /**
+     * Calculate the estimated shipping price for a group of variants
+     */
+    public function estimateShipping(array $payload)
```

Comment:
```suggestion
    public function estimateShipping(array $payload): Response
```

Reply:
All the responses are just objects. Long term I want to make them value objects so you know exactly what you are getting but I think it might take too long right now and with things changing fairly quickly still I didn't want to have to make sure we were updating both places all the time.

I did consider returning the Response itself and then letting the user to `toArray()` or `toObject()` but then I think it would be harder to have nice value objects in the future.